pathway-analysis/setup/exercises/01-pdx_MB_DGE.Rmd

```diff
@@ -0,0 +1,106 @@
+---
+title: "Prepare medulloblastoma PDX DGE results"
+output: 
+  html_notebook: 
+    toc: true
+    toc_float: true
+author: CCDL for ALSF
+date: 2021
+---
+
+This notebook prepares DGE results for the ORA and GSEA pathway analysis exercises.
+
+## Set up
+
+### Libraries
+
+```{r}
+# Pipes
+library(magrittr)
+# DGE analysis and transformation
+library(DESeq2)
+```
+
+### Directories and files
+
+```{r}
+input_dir <- "/shared/data/training-modules/RNA-seq/data/medulloblastoma"
+txi_file <- file.path(input_dir, "txi", "medulloblastoma_txi.RDS")
+```
+
+```{r}
+output_dir <- "/shared/data/training-modules/pathway-analysis/data/medulloblastoma"
+dir.create(output_dir, recursive = TRUE, showWarnings = FALSE)
+dge_file <- file.path(output_dir, "pdx_medulloblastoma_treatment_dge.tsv")
+```
+
+## Read in data
+
+```{r}
+txi <- readr::read_rds(txi_file)
+```
+
+### Dataset creation
+
+```{r ddset-creation}
+# We have Group 3 and Group 4 MB PDX that were treated with digoxin (or control)
+# We're going to use the treatment results for the exercise, so we put that
+# *last*
+ddset <- DESeq2::DESeqDataSet(txi,
+                              design = ~ Group + Treatment)
+```
+
+We need to collapse technical replicates.
+
+```{r}
+ddset <- DESeq2::collapseReplicates(ddset,
+                                    groupby = ddset$mouse_id)
+```
+
+We want the reference to be the untreated/control samples, check levels to make sure that's the case. 
+
+```{r}
+levels(ddset$Treatment)
+```
+
+We need to relevel.
+
+```{r}
+ddset$treatment <- relevel(ddset$Treatment, ref = "none")
+```
+
+## DGE
+
+```{r}
+ddset <- DESeq(ddset)
+```
+```{r}
+resultsNames(ddset)
+```
+
+We used the design `~ Group + Treatment` when creating the `DESeqDataSet`.
+The results for `Treatment` in our example should be the differences between digoxin vs. none, controlling for differences due to molecular subtype (`Group`).
+
+```{r}
+results_lfc <- lfcShrink(ddset, 
+                         coef = "Treatment_none_vs_digoxin",
+                         type = "ashr")
+```
+
+```{r}
+results_df <- results_lfc %>%
+  data.frame() %>%
+  tibble::rownames_to_column("Gene") %>%
+  # We didn't do any prefiltering to this data set, so there are genes 
```

Comment:
Should we do the filtering before we run the DGE and lfc shrinkage steps? I don't recall how much having low counts genes would or would not change these results.

Reply:
DESeq2 does filtering, removing the low count genes mostly helps with saving on memory (not a concern in prepping this), per [the docs on pre-filtering](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#pre-filtering)

> While it is not necessary to pre-filter low count genes before running the DESeq2 functions, there are two reasons which make pre-filtering useful: by removing rows in which there are very few reads, we reduce the memory size of the dds data object, and we increase the speed of the transformation and testing functions within DESeq2

Reply:
Lol. Was just writing a comment with the link to this quote when you beat me to it.